src/main/java/com/snowflake/kafka/connector/internal/streaming/StreamingClientHandler.java

```diff
@@ -50,34 +52,61 @@ public static boolean isClientValid(SnowflakeStreamingIngestClient client) {
     return client != null && !client.isClosed() && client.getName() != null;
   }
 
+  /**
+   * Gets the Properties from the input connector config
+   *
+   * @param connectorConfig configuration properties for a connector
+   * @return the Properties object needed for client creation
+   */
+  public static Properties getClientProperties(Map<String, String> connectorConfig) {
+    Properties streamingClientProps = new Properties();
+
+    if (connectorConfig != null) {
+      streamingClientProps.putAll(
+          StreamingUtils.convertConfigForStreamingClient(new HashMap<>(connectorConfig)));
```

Comment:
Do you know why we need to create a new HashMap?

Reply:
Also should we just return a Properties from `StreamingUtils.convertConfigForStreamingClient`

Reply:
Yes that would be cleaner, altered the convertConfigForStreamingClient to return Properties instead of a map. Removed hashmap clone as i don't see why we need that as well